src/shared/services/activity/activity-socket.service.ts

```diff
@@ -0,0 +1,108 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject, Observable } from 'rxjs';
+import { distinctUntilChanged, filter } from 'rxjs/operators';
+import { Socket } from 'socket.io-client';
+
+import { CaseActivityInfo } from '../../domain/activity';
+import { SessionStorageService } from '../session/session-storage.service';
+import { ActivityService } from './activity.service';
+import { SetCaseActivity, WatchCases } from './models';
+import { Utils } from './utils';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ActivitySocketService {
+  public static SOCKET_MODES: string[] = [ Utils.MODES.socket, Utils.MODES.socketLongPoll ];
+
+  public activity: Observable<CaseActivityInfo[]>;
+  public connect: Observable<any>;
+  public connect_error: Observable<any>;
+  public disconnect: Observable<any>;
+  public connected: BehaviorSubject<boolean> = new BehaviorSubject<boolean>(false);
+
+  public socket: Socket;
+  private pUser: object;
+  public get user(): object {
+    return this.pUser || this.setupUser();
+  }
+
+  public get isEnabled(): boolean {
+    return this.activityService.isEnabled && !!this.socket;
+  }
+
+  constructor(
+    private readonly sessionStorageService: SessionStorageService,
+    private readonly activityService: ActivityService
+  ) {
+    this.activityService.modeSubject
+      .pipe(filter(mode => !!mode))
+      .pipe(distinctUntilChanged())
+      .subscribe(mode => {
+        this.destroy();
+        if (ActivitySocketService.SOCKET_MODES.indexOf(mode) > -1) {
+          this.init();
+        }
+      });
+  }
+
+  public watchCases(caseIds: string[]): void {
+    const payload: WatchCases = { caseIds };
```

Comment:
Surely this is better inline `this.socket.emit('watch', {caseIds})`

Reply:
Agreed. I did this when I thought there'd be more required in the call and just never switched it to something simpler. Doing that now. The same for the other similar methods below.

Reply:
Done.